build.gradle.kts

```diff
@@ -0,0 +1,214 @@
+}
+
+val releaseVersion = releaseVersion()
+val releaseNotes = releaseNotes()
+val isCI = (System.getenv("CI") ?: "false").toBoolean()
+
+group = "org.gradle.android"
+version = releaseVersion.get()
+description = "A Gradle plugin to fix Android caching problems"
+
+repositories {
+    google()
+    mavenCentral()
+}
+
+dependencies {
+    val versions = mapOf(
+        "agp" to "7.3.1",
+        "sdkBuildTools" to "30.0.4",
+        "spock" to "2.3-groovy-3.0",
+    )
+
+    compileOnly(gradleApi())
+    compileOnly("com.android.tools.build:gradle:${versions["agp"]}")
+    compileOnly("com.android.tools:common:${versions["sdkBuildTools"]}")
+    compileOnly("com.android.tools:sdk-common:${versions["sdkBuildTools"]}")
+    implementation("com.google.guava:guava:31.1-jre")
+
+
+    testImplementation(gradleTestKit())
+    testImplementation("com.android.tools.build:gradle:${versions["agp"]}")
+    testImplementation(platform("org.spockframework:spock-bom:${versions["spock"]}"))
+    testImplementation("org.spockframework:spock-core") { exclude(group = "org.codehaus.groovy") }
+    testImplementation("org.spockframework:spock-junit4") { exclude(group = "org.codehaus.groovy") }
+    testImplementation("org.junit.jupiter:junit-jupiter-api")
+}
+
+wrapperUpgrade {
+    gradle {
+        create("android-cache-fix-gradle-plugin") {
+            repo.set("gradle/android-cache-fix-gradle-plugin")
+        }
+    }
+}
+
+java {
+    toolchain {
+        // AGP 7+ only supports JDK 11+
+        languageVersion.set(JavaLanguageVersion.of(11))
+    }
+}
+
+// Main plugin publishing metadata
+gradlePlugin {
+    website.set("https://github.com/gradle/android-cache-fix-gradle-plugin")
+    vcsUrl.set("https://github.com/gradle/android-cache-fix-gradle-plugin")
+
+    plugins {
+        create("androidCacheFixPlugin") {
+            id = "org.gradle.android.cache-fix"
+            displayName = "Gradle Android cache fix plugin"
+            description = releaseNotes.get()
+            implementationClass = "org.gradle.android.AndroidCacheFixPlugin"
+            tags.addAll("android", "cache", "fix")
+        }
+    }
+}
+
+// A local repo we publish our library to for testing in order to workaround limitations
+// in the TestKit plugin classpath.
+val localRepo = file("$buildDir/local-repo")
+
+val isProdPortal = System.getProperty("gradle.portal.url") != null
+// The legacy groupId gradle.plugin.* is only allowed when the plugin
+// has already been published
+val pluginGroupId = if (isCI && isProdPortal) "gradle.plugin.org.gradle.android" else project.group
+publishing {
+    publications {
+        create<MavenPublication>("pluginMaven") {
+            groupId = pluginGroupId.toString()
+        }
+    }
+    repositories {
+        maven {
+            url = uri(localRepo.toURI())
+        }
+    }
+}
+
+// Configuration common to all test tasks
+tasks.withType<Test>().configureEach {
+    dependsOn(tasks.publish)
```

Comment:
can you double check that none of these tasks are configured eagerly? I seem to remember that calling it in this way `tasks.publish` would configure the task eagerly.

Reply:
This is the code for the generated accessor:

![image](https://user-images.githubusercontent.com/5797900/234602356-8564d0c6-f4fd-49c3-abb8-ace54e014230.png)

Which makes it the equivalent of:

```kotlin
tasks.withType<Test>().configureEach {
    dependsOn(tasks.named("publish"))
    // ...
}
```

So, there should be no eager task creation given that lazy APIs are being used.

Reply:
Thanks for checking!